samples/matter/lock/src/app_event.h

```diff
@@ -27,7 +27,7 @@ struct AppEvent {
 
 	AppEvent() = default;
 	AppEvent(LockEventType type, BoltLockManager::OperationSource source) : Type(type), LockEvent{ source } {}
-	explicit AppEvent(FunctionEventType type) : Type(type) {}
+	AppEvent(FunctionEventType type, uint8_t buttonNumber) : Type(type), FunctionEvent{ buttonNumber } {}
```

Comment:
I was considering it, but it would mean duplicating some methods with quite similar logic like to handle press/release events, trigger timer and etc, so I thought maybe adding a parameter allows to avoid it. If you think because of that code became too complex and is less readable I can split it.

Reply:
I thought that way we could keep the connection between button numbers and their functions in a single place, but I don't have a strong opinion on that, so if no one else raises this point we can leave it as it is for now :)